projects/observability/src/shared/components/gauge/basic-gauge/basic-gauge.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input } from '@angular/core';
+import { Color } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-basic-gauge',
+  template: `
+    <g
+      class="input-data"
+      htTooltip="{{ this.value }} of {{ this.maxValue }}"
+    >
+      <path
+        class="value-ring"
+        *ngIf="this.hasRadius"
+        [attr.d]="this.valueArc"
+        [attr.fill]="this.defaultColor"
+      ></path>
+      <text x="0" y="-4" class="value-display" [attr.fill]="this.defaultColor">
+        {{ this.value }}
+      </text>
+      <text x="0" y="20" class="label-display">{{ this.defaultLabel }}</text>
+    </g>
+  `,
+  styleUrls: ['./basic-gauge.component.scss']
+})
+export class BasicGaugeComponent {
+
+  @Input()
+  public value: number = 0;
+
+  @Input()
+  public maxValue: number = 10;
+
+  @Input()
+  public defaultLabel: string = '';
```

Comment:
may be just label and color. I would think most of the logic can live in this component right? The other component can take thresholds and assign it the value, maxValue, label and color to this component. hasRadius, valueArc should be internal to basic(may be a better name) component

Reply:
I removed the file because I couldn't find a way to do the subdivision